syntax = "proto3";
package compare_serialization;
import "google/protobuf/any.proto"; // 引⼊ any.proto ⽂件

// 地址
message Address{
  string home_address = 1; // 家庭地址
  string unit_address = 2; // 单位地址
}

// 联系⼈
message PeopleInfo {
  string name = 1; // 姓名
  int32 age = 2; // 年龄
  message Phone {
    string number = 1; // 电话号码
    enum PhoneType {
      MP = 0; // 移动电话
      TEL = 1; // 固定电话
    }
    PhoneType type = 2; // 类型
  }
  repeated Phone phone = 3; // 电话
  google.protobuf.Any data = 4;
  oneof other_contact { // 其他联系⽅式：多选⼀
    string qq = 5; 
    string weixin = 6;
  }
  map<string, string> remark = 7; // 备注
}
